code/game/objects/structures/dropship_ammo.dm

```diff
@@ -211,6 +210,52 @@
 	T.ignite(5, 30) //short but intense
 
 
+//railgun
+
+/obj/structure/ship_ammo/energy/railgun
+	name = "Railgun ammunition"
+	icon_state = "railgun_ammo"
+	desc = "Changeable capacitors and projectiles clip in single unit, complexity of the design does not allow transfer of ammo. Used to fire railgun."
+	travelling_time = 2 SECONDS
+	equipment_type = /obj/structure/dropship_equipment/weapon/railgun
+	transferable_ammo = FALSE
+	ammo_used_per_firing = 20 // 5 shots
+	point_cost = 200
+	impact_radius = 0
+	ammo_type = CAS_RAILGUN
+
+/obj/structure/ship_ammo/energy/railgun/examine(mob/user)
+	. = ..()
+	to_chat(user, "It's at [round(100*ammo_count/max_ammo_count)]% charge and [ammo_count/20] projectiles.")
+
+
+/obj/structure/ship_ammo/energy/railgun/show_loaded_desc(mob/user)
+	if(ammo_count)
+		to_chat(user, "It's loaded with \a [src] at [round(100*ammo_count/max_ammo_count)]% charge and [ammo_count] projectiles.")
+	else
+		to_chat(user, "[name] is empty.")
+
+///Defining projectlie for versatile damage config and enabling bullet_act()
+/datum/ammo/bullet/cas_railgun
+
+/datum/ammo/bullet/cas_railgun/on_hit_mob()
+
+///Rail ammo acts on the turf passed in
+/obj/structure/ship_ammo/energy/railgun/projectile_impact(turf/T)
```

Comment:
Still trying to find best damaging code, not final. Any suggestion will be appreciated.